tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/TeamMailboxMemberSetMethod.scala

```diff
@@ -0,0 +1,138 @@
+import org.apache.james.metrics.api.MetricFactory
+import org.apache.james.user.api.UsersRepository
+import org.apache.james.util.ReactorUtils
+import org.reactivestreams.Publisher
+import play.api.libs.json.JsObject
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class TeamMailboxMemberSetMethod @Inject()(val teamMailboxRepository: TeamMailboxRepository,
+                                           val usersRepository: UsersRepository,
+                                           val metricFactory: MetricFactory,
+                                           val sessionTranslator: SessionTranslator,
+                                           val sessionSupplier: SessionSupplier) extends MethodRequiringAccountId[TeamMailboxMemberSetRequest] {
+
+  override val methodName: Invocation.MethodName = MethodName("TeamMailboxMember/set")
+
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, JMAP_MAIL, LINAGORA_TEAM_MAILBOXES)
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, TeamMailboxMemberSetRequest] =
+    Serializer.deserializeSetRequest(invocation.arguments.value)
+      .asEither.left.map(ResponseSerializer.asException)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier],
+                         invocation: InvocationWithContext,
+                         mailboxSession: MailboxSession,
+                         request: TeamMailboxMemberSetRequest): Publisher[InvocationWithContext] =
+    performUpdates(mailboxSession.getUser, request)
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(Serializer.serializeSetResponse(response).as[JsObject]),
+        methodCallId = invocation.invocation.methodCallId))
+      .map(InvocationWithContext(_, invocation.processingContext))
+
+  private def performUpdates(username: Username, request: TeamMailboxMemberSetRequest): SMono[TeamMailboxMemberSetResponse] =
+    SFlux.fromIterable(request.validatedUpdateRequest())
+      .flatMap(parsingRequestResult => performUpdate(username, parsingRequestResult), ReactorUtils.LOW_CONCURRENCY)
+      .collectSeq()
+      .map(listResult => TeamMailboxMemberSetResponse.from(request.accountId, listResult))
+
+  private def performUpdate(username: Username, parsingRequestResult: TeamMailboxMemberParsingRequestResult): SMono[TeamMailboxMemberSetResult] =
+  {
+    val r: SMono[TeamMailboxMemberSetResult] = for {
+      requestWithValidSyntax <- validateSyntax(parsingRequestResult)
+      requestWithUserExistenceValidation <- validateUsersExist(requestWithValidSyntax)
+      presentMembers <- retrievePresentMembers(requestWithUserExistenceValidation)
+      requestWithAuthorizationValidation <- validateAuthorizations(username, requestWithUserExistenceValidation, presentMembers)
+      requestWithAuthorizationValidation <- validateNoChangeToManagers(requestWithAuthorizationValidation, presentMembers)
```

Comment:
a bit confuse,  when see double variable has same name `requestWithAuthorizationValidation` 
it will better:
`requestWithAuthorizationValidation` -> `validatedRequest`